### PR TITLE
Add super gaussian laser

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR adds a super-gaussian laser profile and updates the documentation

### CODE REVIEW

File: lasy/profiles/transverse/super_gaussian_profile.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        ----------
+        w0: float (in meter)
+            The waist of the laser pulse, i.e. :math:`w_0` in the above formula.
+
+        n_order: float (in meter)
+            The shape parameter of the super-gaussian function, i.e. :math:`n` in the above formula.
+            If :math:`n=2` the super-Gaussian becomes a standard Gaussian function.
+            If :math:`n=1` the super-Gaussian becomes a Laplace function.
+        """
+        self.w0 = w0
```
Comment (MaxThevenet) [inline lasy/profiles/transverse/super_gaussian_profile.py:30]: As (pretty neatly) suggested by codeQL, I think we should call the base Classe's `__init__`
```suggestion
        super().__init__()
        self.w0 = w0
```

File: docs/source/api/profiles/transverse/super_gaussian_profile.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Super-Gaussian Transverse Profile
+===================================
+
+Used to define a super-Gaussian transverse laser profile.
+
+Super-Gaussian profiles are modified Gaussian-like profiles with a flat-top and a Gaussian fall-off.
+The shape of the profile is characterised by the waist :math:`w_0` and by one "order parameter" :math:`n`.
```
Comment (MaxThevenet) [inline docs/source/api/profiles/transverse/super_gaussian_profile.rst]:
```suggestion
Used to define a super-Gaussian transverse laser profile.

The shape of the profile is characterised by the waist :math:`w_0` and by one "order parameter" :math:`n`, where :math:`n=2` gives a standard Gaussian profile, and the profile converges to a flat-top when :math:`n` goes to infinity.
```

File: docs/source/api/profiles/transverse/super_gaussian_profile.rst [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Super-Gaussian Transverse Profile
+===================================
```
Comment (MaxThevenet) [inline docs/source/api/profiles/transverse/super_gaussian_profile.rst]:
```suggestion
=================================
```

File: tests/test_laser_profiles.py [hunk 4]
```diff
@@ -70,3 +71,29 @@ def test_profile_laguerre_gauss():
...
+
+def test_profile_super_gauss():
+    # Case with super-Gaussian laser
+    wavelength=.8e-6
+    pol = (1,0)
+    laser_energy = 1. # J
+    t_peak = 0.e-15 # s
+    tau = 30.e-15 # s
+    w0 = 5.e-6 # m
+    beta = 10.0 # m
```
Comment (MaxThevenet) [inline tests/test_laser_profiles.py]: betais not used. Instead, `n_order` is passed directly, which works well too.
```suggestion
```

**Review by MaxThevenet — CHANGES_REQUESTED** [review]
Looks great, thanks for this PR! See comments and suggestions below.

File: lasy/profiles/transverse/super_gaussian_profile.py (hunk 1 above)
Comment (lucafedeli88) [reply lasy/profiles/transverse/super_gaussian_profile.py:30]: good idea!

**Comment by lucafedeli88** [comment]
> Looks great, thanks for this PR! See comments and suggestions below.

thanks for the review!

**Review by MaxThevenet — APPROVED** [review]
Thanks!

**Pull request merged**